src/components/SideBar/SideBar.jsx

```diff
@@ -1,2 +1,83 @@
-import React from "react";
-export const SideBar = () => <>SideBar</>;
+import { useState, useRef, useEffect } from "react";
+import {
+  StyledText,
+  StyledSidebar,
+  StyledLink,
+  StyledTopWrapper,
+  StyledLogoSvg,
+  StyledLogoText,
+  StyledBarsOutlined,
+  StyledSvgWrapper,
+  StyledSideBarContainer,
+  StyledInputWrapper,
+  StyledSearch,
+  StyledSearchBtn,
+} from "./style";
+import { menuItemsArr } from "./menuItems";
+import { SearchOutlined } from "@ant-design/icons";
+
+export const SideBar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleSidebar = () => setIsOpen(!isOpen);
+  const menuItems = menuItemsArr;
```

Comment:
Why do you copy link to an array locally? It will work with same array no matter what